tests/unit/s2n_ktls_io_test.c

```diff
@@ -286,5 +300,217 @@ int main(int argc, char **argv)
         };
     };
 
+    /* Test s2n_ktls_recvmsg */
+    {
+        /* Safety */
+        {
+            DEFER_CLEANUP(struct s2n_connection *client = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            uint8_t recv_buf[S2N_TLS_MAXIMUM_FRAGMENT_LENGTH] = { 0 };
+            s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+            uint8_t recv_record_type = 0;
+            size_t bytes_read = 0;
+
+            EXPECT_ERROR_WITH_ERRNO(
+                    s2n_ktls_recvmsg(NULL, &recv_record_type, recv_buf, S2N_TEST_TO_SEND, &blocked, &bytes_read),
+                    S2N_ERR_NULL);
+            EXPECT_ERROR_WITH_ERRNO(
+                    s2n_ktls_recvmsg(client, &recv_record_type, NULL, S2N_TEST_TO_SEND, &blocked, &bytes_read),
+                    S2N_ERR_NULL);
+            EXPECT_ERROR_WITH_ERRNO(
+                    s2n_ktls_recvmsg(client, &recv_record_type, recv_buf, S2N_TEST_TO_SEND, NULL, &bytes_read),
+                    S2N_ERR_NULL);
+            EXPECT_ERROR_WITH_ERRNO(
+                    s2n_ktls_recvmsg(client, &recv_record_type, recv_buf, S2N_TEST_TO_SEND, &blocked, NULL),
+                    S2N_ERR_NULL);
```

Comment:
missing &recv_record_type -> NULL

Reply:
recv_record_type is not actually used in the fn(its passed to s2n_ktls_get_control_data) and therefore there is no NULL check at the beginning.

We already NULL for that fn [here](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_ktls_io_test.c#L71-L72).

I dont think we want to enforce safety (since its not used) and therefore I dont think we want to test it either.

Reply:
If you're going to base your safety tests on the actual implementation, you're going to miss things. This is particularly true if you write a method that ends up passing an input on to an older method without good null checking. Safety tests are very cheap-- be thorough.